src/Services/SignatureFactory.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace LearnositySdk\Services;
+
+use LearnositySdk\Services\Signatures\SignatureInterface;
+use LearnositySdk\Services\Signatures\HashSignature;
+use LearnositySdk\Services\Signatures\HmacSignature;
+
+class SignatureFactory
+{
+    const LEGACY_SIGNATURE_VERSION = '01';
+
+    const DEFAULT_SIGNATURE_VERSION = '02';
+    /**
+     * @param string $version
+     * @return SignatureInterface
+     * @throws \Exception
+     */
+    public static function getSignatureGenerator(string $version): SignatureInterface
```

Comment:
We should allow not specifying a version here, so the factory controls the default.

Reply:
Hmm, I don't know. Feels like the factory is in charge of building SignatureGenerators as requested. The fact that currently we want v2 to be the default feels like a business rule of the SDK that might change in the future, not something the Factory needs to be aware of.

Reply:
Ok Ive left in for now. Let me know if that is ok.